tests/integ/test_dataframe.py

```diff
@@ -387,6 +406,96 @@ def process(self, n: int) -> int:
         df.select(two_x_udtf("a"), "b", two_x_udtf("c"))
     assert "At most one table function can be called" in str(ex_info)
 
+    @udtf(output_schema=["two_x", "three_x"])
+    class multiplier_udtf:
+        def process(self, n: int) -> Iterable[Tuple[int, int]]:
+            yield (2 * n, 3 * n)
+
+    with pytest.raises(ValueError) as ex_info:
+        df.select(multiplier_udtf(df.a).alias("double", "double"))
+    assert "All output column names after aliasing must be unique" in str(ex_info)
+
+    with pytest.raises(ValueError) as ex_info:
+        df.select(multiplier_udtf(df.a).alias("double"))
+    assert (
+        "The number of aliases should be same as the number of cols added by table function"
+        in str(ex_info)
+    )
+
+
+def test_with_column(session):
+    df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
```

Comment:
Probably also worth adding tests for built-in table functions

Reply:
Are there any built-in functions that have a single column output?

Reply:
Probably this one? https://docs.snowflake.com/en/sql-reference/functions/generator.html (or you can search in all snowflake table functions https://docs.snowflake.com/en/sql-reference/functions-table.html)

Reply:
Then is it possible to support `df.with_columns(["a", "b", "c"], [col("x"), my_udtf("y", "z")])` where `my_udtf` returns 2 columns?

Reply:
the current version of the code does not support that. I could try to make it work

Reply:
updated code to make it work like you suggested. also added tests that check this

Reply:
nice